src/inmanta/agent/forking_executor.py

```diff
@@ -440,61 +440,80 @@ def can_be_cleaned_up(self, retention_time: int) -> bool:
     def last_used(self) -> datetime.datetime:
         return self.connection.last_used_at
 
-    async def force_stop(self, grace_time: float = inmanta.const.SHUTDOWN_GRACE_HARD) -> None:
-        """Stop by process close"""
+    async def force_stop(self, grace_time: float = inmanta.const.EXECUTOR_GRACE_HARD) -> None:
+        """
+        Stop by process close
+
+        This method will never raise an exeption, but log it instead.
+        """
         self.closing = True
         await asyncio.get_running_loop().run_in_executor(
             self.owner.thread_pool, functools.partial(self._force_stop, grace_time)
         )
 
     def _force_stop(self, grace_time: float) -> None:
+        """This method will never raise an exeption, but log it instead, as it is used as a finalizer"""
         if self.closed:
             return
         try:
-            self.process.terminate()
-            self.process.join(grace_time)
-            self.process.kill()
-            self.process.join()
-        except ValueError:
-            # Process already gone:
-            pass
-        self._set_closed()
-
-    def _set_closed(self) -> None:
-        # this code can be raced from the join call and the disconnect handler
-        # relying on the GIL to keep us safe
-        if not self.closed:
-            self.closing = True
-            self.closed = True
-            self.process.close()
-            self.owner._child_closed(self)
+            if self.process.exitcode is None:
+                # Running
+                self.process.join(grace_time)
```

Comment:
Because it is not needed. The idea is that we send stop to the remote end, which drops out of the main loop. 
This should cause a clean shutdown.

Here we just introduce a race, calling in the stop procedure over two distinct channels (IPC and a term signal)

Reply:
Could you point me to the other point where we stop it?